src/NLogTarget/ApplicationInsightsTarget.cs

```diff
@@ -49,6 +50,37 @@ internal TelemetryClient TelemetryClient
             get { return this.telemetryClient; }
         }
 
+        internal void BuildPropertyBag(LogEventInfo logEvent, ITelemetry trace)
+        {
+            trace.Timestamp = logEvent.TimeStamp;
+            trace.Sequence = logEvent.SequenceID.ToString(CultureInfo.InvariantCulture);
+
+            IDictionary<string, string> propertyBag;
+
+            if (trace is ExceptionTelemetry)
```

Comment:
Would it help to use the `ISupportProperties` interface here? i.e.
```C#
propertyBag = ((ISupportProperties)trace).Properties;
```

Reply:
I'm just moving code around to fix the unit tests.

Reply:
Ah, I see. Fine to leave it as it is, then.